lightning/src/routing/network_graph.rs

```diff
@@ -243,6 +250,122 @@ impl<C: Deref + Sync + Send, L: Deref + Sync + Send> RoutingMessageHandler for N
 			false
 		}
 	}
+
+	/// Initiates a stateless sync of routing gossip information with a peer
+	/// by calling query_channel_range. We should expect one or more
+	/// reply_channel_range messages in response to our query. Each reply
+	/// will enqueue a query_scid message to request gossip messages for each
+	/// channel. The sync is considered complete when the final reply_scids_end
+	/// message is received, though we are not tracking this directly.
+	fn sync_channel_range(&self, their_node_id: &PublicKey, chain_hash: BlockHash, first_blocknum: u32, number_of_blocks: u32) {
```

Comment:
Right, so this is still a bit verbose.

As for `chain_hash`, I think just make the `NetworkGraph` aware of the current chain hash and take it in the constructor or so. We probably want to do this anyway because we should check the chain_hash in `handle_channel_announcement` even if we don't have a `chain_access` filled in.

In general, though, I'm just not quite sure what the right answer is here - it seems wrong for the peer handler (something which is really meant to just be a dumb multiplexer of events and messages across handlers and peers) to know anything about the current chain height, either. Instead, a simple "peer has connected, their_node_id is X" API seems like what `PeerHandler` wants, though what we do from there is murky - making NetGraphMsgHandler aware of the current chain also sucks.

Do you think it makes sense to use that information to do a sync from all peers that starts from block `95th percentile of the channels we have`? It would be vulnerable to fake channel announcement attacks, but maybe that's OK? Obviously long-term we should be thinking about using set-reconciliation to fix this, but that's probably a ways off.